components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java

```diff
@@ -944,28 +945,32 @@ private boolean isCustomInboundAuthType(String authType) {
     }
 
     /**
-     * Get all custom authenticators
+     * Get all custom authenticators.
      *
      * @return Custom authenticators
      */
     public List<InboundAuthenticationRequestConfig> getInboundAuthenticators() {
 
-        if (!CollectionUtils.isEmpty(inboundAuthenticationRequestConfigs)) {
-            return inboundAuthenticationRequestConfigs;
-        }
-
-        inboundAuthenticationRequestConfigs = new ArrayList<InboundAuthenticationRequestConfig>();
-
-        InboundAuthenticationRequestConfig[] authRequests = serviceProvider
-                .getInboundAuthenticationConfig()
-                .getInboundAuthenticationRequestConfigs();
-
-        if (authRequests != null) {
-            for (InboundAuthenticationRequestConfig request : authRequests) {
-                if (isCustomInboundAuthType(request.getInboundAuthType())) {
-                    inboundAuthenticationRequestConfigs.add(request);
+        if (customInboundAuthenticatorConfigs != null && customInboundAuthenticatorConfigs.size() > 0) {
```

Comment:
we can use CollectionUtils.isNotEmpty(customInboundAuthenticatorConfigs) && CollectionUtils.isNotEmpty(inboundAuthenticationRequestConfigs) here and remove nested if

Reply:
If we remove the nested if condition and use a &&, we have to do the CollectionUtils.isNotEmpty(customInboundAuthenticatorConfigs)  check twice.

Reply:
where do we check CollectionUtils.isNotEmpty(customInboundAuthenticatorConfigs)  for the second time? we can remove the second check

Reply:
If we remove the nested if, we have to check twice.